### PR TITLE
Adds a cap to Xenobiology Mobs and Allows Sentient Giant Spider prompts to be disabled

### PR DESCRIPTION
**Pull request by Sirryan2002**
## What Does This PR Do
- Changes cap introduced in #15923 to xenobiology specific giant spiders only (Eggs stop growing when total mobs>cap-10 and spiderlings die instead of spawning spiders when total mobs=cap)
- Adds a mob cap to xenobiology mobs (Currently set at 45 & Does not include slimes). Spawning xenobio mobs from gold cores increases counter, said mobs dying decreases the counter.
- Sets giant spiders prompt flag to ROLE_SENTIENT instead of ROLE_GSPIDER so those who have sentient mobs disabled do not get prompted

## Why It's Good For The Game
Admins are tired of coming up with IC methods/reasoning to cleaning up xenobio mob explosions and it usually just ends with players being confused in dchat as to why we leveled xenobiology or deleted all of their mobs.

Sentient Giant spiders dchat prompts can now be disabled by disabling the Sentient Mobs preference in game preferences. Getting spammed with 40 dings and prompts over the course of 5-10 minutes is annoying and should be avoidable. In a future PR (or if someone wants to help me), the preference for giant spiders in game preferences should be labeled as `terror spider` since giant spiders only prompted naturally from xenobiology before and this flag only applies to Terrors now.

This PR will prevent significant loads being placed on SSmobs by xenobiology. I think 45 mobs is generous so feel free to weigh in your opinion if you think I should drop it down to 40 or even 30.

This was tested by spawning in xenobio mobs with gold slime cores. I don't have the skill to set up a two client local server so I'm not 100% sure how sentience potions will affect the `xenobiology_spawned` variable but it should be fine. Testing on local server produced no runtimes and as long as the initial nurse spider has the `xenobiology_spawned` var set to `TRUE` then all of it's children will also have the var set to true.
## Changelog
:cl:
add: Adds a 45 mob cap on spawning xenobiology mobs
tweak: Sentient Giant Spider dchat prompts now fall under the sentient mob preference
/:cl:

### CODE REVIEW

File: code/game/objects/effects/spiders.dm [hunk 1]
```diff
@@ -1,6 +1,4 @@
-//generic procs copied from obj/effect/alien
-#define SPIDER_SOFT_CAP 30
-#define SPIDER_HARD_CAP 40
+
```
Comment (dearmochi) [inline code/game/objects/effects/spiders.dm:1]:
```suggestion
```

File: code/game/objects/effects/spiders.dm [hunk 2]
```diff
@@ -174,19 +177,22 @@
...
 		if(amount_grown >= 100)
-			if(SSmobs.giant_spiders > SPIDER_HARD_CAP)
+			if(SSmobs.xenobiology_mobs > MAX_GOLD_CORE_MOBS && xenobiology_spawned)
 				qdel(src)
 				return
 			if(!grow_as)
 				grow_as = pick(typesof(/mob/living/simple_animal/hostile/poison/giant_spider))
 			var/mob/living/simple_animal/hostile/poison/giant_spider/S = new grow_as(loc)
-			SSmobs.giant_spiders++
+			SSmobs.xenobiology_mobs++
```
Comment (dearmochi) [inline code/game/objects/effects/spiders.dm:186]:
```suggestion
```

You're incrementing it already a few lines below, accidental?

File: code/game/objects/effects/spiders.dm [hunk 3]
```diff
@@ -174,19 +177,22 @@
...
 			S.faction = faction.Copy()
 			S.master_commander = master_commander
+			S.xenobiology_spawned = xenobiology_spawned
+			if (xenobiology_spawned)
```
Comment (dearmochi) [inline code/game/objects/effects/spiders.dm:190]:
```suggestion
			if(xenobiology_spawned)
```

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 4]
```diff
@@ -374,6 +376,8 @@
 			visible_message("<span class='danger'>\The [src] [deathmessage]</span>")
 		else if(!del_on_death)
 			visible_message("<span class='danger'>\The [src] stops moving...</span>")
+	if (xenobiology_spawned)
```
Comment (dearmochi) [inline code/modules/mob/living/simple_animal/simple_animal.dm:379]:
```suggestion
	if(xenobiology_spawned)
```

File: code/modules/reagents/chemistry/recipes.dm [hunk 5]
```diff
@@ -48,6 +48,8 @@
 				S = create_random_mob(get_turf(holder.my_atom), mob_class)
 			else
 				S = new mob_class(get_turf(holder.my_atom))//Spawn our specific mob_class
+			if (gold_core_spawn) //For tracking xenobiology mobs
```
Comment (dearmochi) [inline code/modules/reagents/chemistry/recipes.dm:51]:
```suggestion
			if(gold_core_spawn) //For tracking xenobiology mobs
```

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 6]
```diff
@@ -120,8 +128,11 @@
 
 /datum/chemical_reaction/slimemobspawn/friendly/summon_mobs(datum/reagents/holder, turf/T)
 	T.visible_message("<span class='danger'>The slime extract begins to vibrate adorably!</span>")
-	addtimer(CALLBACK(src, .proc/chemical_mob_spawn, holder, 1, "Friendly Gold Slime", FRIENDLY_SPAWN, "neutral"), 50)
-
+	if(SSmobs.xenobiology_mobs < MAX_GOLD_CORE_MOBS)
+		addtimer(CALLBACK(src, .proc/chemical_mob_spawn, holder, 1, "Friendly Gold Slime", FRIENDLY_SPAWN, "neutral", TRUE, TRUE), 50)
+		SSmobs.xenobiology_mobs += 1
+	else
+		T.visible_message("<span class='danger'>The slime extract sputters out, there's too many mobs to make any more!</span>")
```
Comment (dearmochi) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm:135]:
```suggestion
		T.visible_message("<span class='danger'>The slime extract sputters out, there's too many mobs to make any more!</span>")

```

**Review by dearmochi — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/spiders.dm (hunk 2 above)
Comment (Sirryan2002) [reply code/game/objects/effects/spiders.dm:186]: yeah, good catch

**Review by dearmochi — APPROVED** [review]

**Pull request merged**